resources/Microsoft.Windows.Setting.Time/Microsoft.Windows.Setting.Time.psm1

```diff
@@ -0,0 +1,235 @@
+    #     "Magallanes Standard Time", "Saint Pierre Standard Time", "Bahia Standard Time", "UTC-02", "Mid-Atlantic Standard Time", 
+    #     "Azores Standard Time", "Cape Verde Standard Time", "UTC", "Morocco Standard Time", "GMT Standard Time", 
+    #     "Greenwich Standard Time", "W. Europe Standard Time", "Central Europe Standard Time", "Romance Standard Time", 
+    #     "Central European Standard Time", "W. Central Africa Standard Time", "Namibia Standard Time", "Jordan Standard Time", 
+    #     "GTB Standard Time", "Middle East Standard Time", "Egypt Standard Time", "E. Europe Standard Time", "Syria Standard Time", 
+    #     "West Bank Standard Time", "South Africa Standard Time", "FLE Standard Time", "Israel Standard Time", "Kaliningrad Standard Time", 
+    #     "Sudan Standard Time", "Libya Standard Time", "Namibia Standard Time", "Arabic Standard Time", "Turkey Standard Time", 
+    #     "Arab Standard Time", "Belarus Standard Time", "Russian Standard Time", "E. Africa Standard Time", "Iran Standard Time", 
+    #     "Arabian Standard Time", "Astrakhan Standard Time", "Azerbaijan Standard Time", "Russia Time Zone 3", "Mauritius Standard Time", 
+    #     "Saratov Standard Time", "Georgian Standard Time", "Caucasus Standard Time", "Afghanistan Standard Time", "West Asia Standard Time", 
+    #     "Ekaterinburg Standard Time", "Pakistan Standard Time", "India Standard Time", "Sri Lanka Standard Time", "Nepal Standard Time", 
+    #     "Central Asia Standard Time", "Bangladesh Standard Time", "Omsk Standard Time", "Myanmar Standard Time", "SE Asia Standard Time", 
+    #     "Altai Standard Time", "W. Mongolia Standard Time", "North Asia Standard Time", "N. Central Asia Standard Time", 
+    #     "Tomsk Standard Time", "China Standard Time", "North Asia East Standard Time", "Singapore Standard Time", "W. Australia Standard Time", 
+    #     "Taipei Standard Time", "Ulaanbaatar Standard Time", "North Korea Standard Time", "Aus Central W. Standard Time", 
+    #     "Transbaikal Standard Time", "Tokyo Standard Time", "Korea Standard Time", "Yakutsk Standard Time", "Cen. Australia Standard Time", 
+    #     "AUS Central Standard Time", "E. Australia Standard Time", "AUS Eastern Standard Time", "West Pacific Standard Time", 
+    #     "Tasmania Standard Time", "Vladivostok Standard Time", "Lord Howe Standard Time", "Bougainville Standard Time", 
+    #     "Russia Time Zone 10", "Magadan Standard Time", "Norfolk Standard Time", "Sakhalin Standard Time", "Central Pacific Standard Time", 
+    #     "Russia Time Zone 11", "New Zealand Standard Time", "UTC+12", "Fiji Standard Time", "Kamchatka Standard Time", 
+    #     "Chatham Islands Standard Time", "UTC+13", "Tonga Standard Time", "Samoa Standard Time", "Line Islands Standard Time"
+    # )]
+    [string] $TimeZone = (Get-TimeZone).Id
+
+    [DscProperty()]
+    [TimeZoneAutomatically] $SetTimeZoneAutomatically = [TimeZoneAutomatically]::NTP
```

Comment:
Why an enum? Couldn’t this just be a boolean?

Reply:
I don't know what it was, but somehow it had odd behaviours when doing so. is it nit or does it needs to be adjusted?

Reply:
Will test on my machine after work - I have a few ideas I need to check

Reply:
So a couple things - On both of my machines, updating this registry key changes the value of `Set time automatically` and not `Set time zone automatically`

Then, when I checked out the PR without any modifications attempting to use the resource failed to set the desired state -
```
PS C:\Users\Trenly> Invoke-DscResource -Name Time -ModuleName Microsoft.Windows.Setting.Time -Method Set -Property @{SetTimeZoneAutomatically = 'NoSync'}

RebootRequired
--------------
         False

PS C:\Users\Trenly> Invoke-DscResource -Name Time -ModuleName Microsoft.Windows.Setting.Time -Method Get -Property @{}

TimeZone              SetTimeZoneAutomatically ShowSystemTrayDateTime
--------              ------------------------ ----------------------
Central Standard Time                      NTP                  False
```

After looking at the registry more, I see why you chose to use the enum, but the registry key is really only acting as a boolean. I did some investigation and found that if the key is not present, the key is empty, or the key has *any value* other than `NoSync` (case insensitive), then NTP is enabled and the time will be set automatically.  

My recommendation would be to have the parameter be a nullable bool, and then add two hidden strings so you can do something like this:
```
static hidden [string] $NtpEnabled= 'NTP'
static hidden [string] $NtpDisabled= 'NoSync'
. . .
if ($currentState.SetTimeAutomatically -ne $this.SetTimeAutomatically)
{
	$desiredState = $this.SetTimeAutomatically ? [Time]::NtpEnabled : [Time]::NtpDisabled
	Set-ItemProperty -Path $global:tzAutoUpdatePath -Name ([Time]::SetTimeZoneAutomaticallyProperty) -Value $desiredState
}
```

That way the user is presented with a boolean option both in settings and in DSC. However, we can wait for @ryfu-msft or @denelon to give their opinions

Reply:
I have added this, can you verify once more?